### ISSUE TITLE
(Re-)implement masked token modeling training objective

### ISSUE THREAD

**Issue by clessig** [ecmwf/WeatherGenerator#42]
### Is your feature request related to a problem? Please describe.

Masked token modeling was removed as training objective. It needs to be (re-)implemented and properly parametrized.

### Additional context

Previous experiments clearly indicated that pre-training for a large number of input streams, masked token modeling with fine-tuning for forecasting leads to better results than training directly for forecasting. This is consistent with theory that suggests that masked token modeling helps to learn correlations between multiple related but distinct inputs.

### Organisation

ECMWF
